File: frontend/src/types/Matricula.ts
export interface Matricula {
  id: number;
  alunoId: number;
  cursoId: number;
  dataMatricula: string;
  dataConclusao?: string;
  status: string;
  notaFinal?: number;
  ativo: boolean;
  // Propriedades de navegação
  alunoNome?: string;
  alunoSobrenome?: string;
  cursoNome?: string;
}

export interface CreateMatriculaRequest {
  alunoId: number;
  cursoId: number;
  dataMatricula?: string;
  status?: string;
  ativo?: boolean;
}

export interface UpdateMatriculaRequest {
  dataConclusao?: string;
  status?: string;
  notaFinal?: number;
  ativo?: boolean;
}

export interface MatriculaResumo {
  id: number;
  alunoId: number;
  alunoNome: string;
  alunoSobrenome: string;
  cursoId: number;
  cursoNome: string;
  dataMatricula: string;
  dataConclusao?: string;
  status: string;
  notaFinal?: number;
  ativo: boolean;
}

File: backend/Api.EscolaIdiomas/Api.EscolaIdiomas.Domain/Models/Matriculas/Matricula.cs
using Api.EscolaIdiomas.Domain.Models.Alunos;
using Api.EscolaIdiomas.Domain.Models.Cursos;

namespace Api.EscolaIdiomas.Domain.Models.Matriculas
{
    public class Matricula
    {
        public long Id { get; set; }
        public long AlunoId { get; set; }
        public long CursoId { get; set; }
        public DateTime DataMatricula { get; set; }
        public DateTime? DataConclusao { get; set; }
        public string Status { get; set; } = "Ativa";
        public decimal? NotaFinal { get; set; }
        public bool Ativo { get; set; }
        
        // Propriedades de navegação (opcional, para facilitar consultas)
        public Aluno? Aluno { get; set; }
        public Curso? Curso { get; set; }
    }
}
